src/submodule.c

```diff
@@ -1916,31 +1921,32 @@ static int submodule_load_from_wd_lite(git_submodule *sm)
 
 /**
  * Returns a snapshot of $WORK_TREE/.gitmodules.
- *
- * We ignore any errors and just pretend the file isn't there.
  */
-static git_config *gitmodules_snapshot(git_repository *repo)
+static int gitmodules_snapshot(git_config **snap, git_repository *repo)
 {
 	const char *workdir = git_repository_workdir(repo);
-	git_config *mods = NULL, *snap = NULL;
-	git_buf path = GIT_BUF_INIT;
 
 	if (workdir != NULL) {
```

Comment:
The whole indentation could be avoided if you return early in case `workdir == NULL`. That's a matter of taste, though -- your call if you want to do that instead.

Reply:
Yeah I also thought that, however, I did not want to make too big changes...